squad/core/utils.py

```diff
@@ -18,8 +18,20 @@ def parse_name(input_name):
 
     parts = input_name.split('/')
     if len(parts) > 1:
-        group_name = '/'.join(parts[0:-1])
-        name = parts[-1]
+        # if test name is in format suite/complex.name[some/test/variant]
+        # don't split the name in brackets
+        # this is workaround for some CTS test names
+        if parts[-1].endswith("]"):
+            index = len(parts) - 1
+            # find index of the part that contains opening bracket
+            for part in reversed(parts):
+                if "[" in part:
+                    index = parts.index(part)
+            group_name = '/'.join(parts[0:index])
+            name = '/'.join(parts[index:])
+        else:
+            group_name = '/'.join(parts[0:-1])
+            name = parts[-1]
```

Comment:
I was skeptical that this algorithm worked in all cases but I added some tests and couldn't find any bugs (see https://github.com/danrue/squad/commit/b1df44c547e52c760d397d977c42287682b623b2 if you want to add them). Even so, I think it might have be simpler to pull out the [string] using a regex or similar rather than a loop.